kirin/core/handler.py

```
@@ -402,18 +402,25 @@ def get_next_stop():
                 else:
                     # selection of extra iterable stop-times
                     sp_id = new_stu.stop_id
-                    if is_new_stop_event_valid(event_name='arrival', stop_id=sp_id, stop_order=order,
+                    if is_new_stop_event_valid(event_name='arrival', stop_id=sp_id.encode('utf-8'), stop_order=order,
```

What about the case where **none** of the `arrival` and `departure` return `true` for `is_new_stop_event_valid` ?
You might want to raise an error in the `else` case for this condition.

---

This is not exactly related the purpose of this PR, but in that case Kirin just skips that stop, and `is_new_stop_event_valid` outputs a warning.